helix-lsp/src/lib.rs

```diff
@@ -420,6 +420,7 @@ pub mod util {
             return transaction;
         }
 
```

Comment:
Can you explain/add a comment why normalize isn't needed here? This is an additional invariant that we should document in places where no_normalize is used

Reply:
I added a comment but this was actually exiting behaviour. Really this isn't a selection we produce here at all, we need the mapped ranges for calculating the actual selecton below (which is normalized later). In the past I achieved that by calling `map_pos' on the ranges directly (below). Normalizing here not only has no point it would also break the association between tabstops and selections (if selections are merged/sorted). I just switched to the selection function here to also take advantage of the optimization here but really this is not a selection at all we are just updating a list of ranges.